src/replication.c

```diff
@@ -1970,7 +1970,7 @@ void replicationCron(void) {
     }
 
     /* Timed out master when we are an already connected slave? */
-    if (server.masterhost && server.repl_state == REDIS_REPL_CONNECTED &&
+	if (server.masterhost && server.repl_state == REDIS_REPL_CONNECTED && server.master &&
```

Comment:
Yes, you are right. This is not Windows-specific. 
The changes belonging to this commit ( ea5e668bc11a332d7084a5be5f421838450e8cbd ) is  not specific to Windows, too. 
Just added null-pointer check in order to remove potential memory-safety risk factors.